src/parser/Expression.ts

```diff
@@ -1290,6 +1290,90 @@ export class AnnotationExpression extends Expression {
     }
 }
 
+export class TernaryExpression extends Expression {
+    constructor(
+        readonly test: Expression,
+        readonly questionMarkToken: Token,
+        readonly consequent: Expression,
+        readonly colonToken: Token,
+        readonly alternate: Expression
+    ) {
+        super();
+        this.range = util.createRangeFromPositions(
+            test.range.start,
+            (alternate ?? colonToken ?? consequent ?? questionMarkToken ?? test).range.end
+        );
+    }
+
+    public range: Range;
+
+    transpile(state: TranspileState) {
+        let result = [];
+        let consequentInfo = util.getExpressionInfo(this.consequent);
+        let alternateInfo = util.getExpressionInfo(this.alternate);
+
+        //get all unique variable names used in the consequent and alternate, and sort them alphabetically so the output is consistent
+        let allUniqueVarNames = [...new Set([...consequentInfo.uniqueVarNames, ...alternateInfo.uniqueVarNames])].sort();
+        let mutatingExpressions = [
+            ...consequentInfo.expressions,
+            ...alternateInfo.expressions
+        ].filter(e => e instanceof CallExpression || e instanceof CallfuncExpression || e instanceof DottedGetExpression);
+
+        if (mutatingExpressions.length > 0) {
+            result.push(
+                `(function(condition, `,
```

Comment:
I think using `condition` for the arg name is risky. Maybe `bslib_cond` will be safer?

Reply:
Very good point. Will address

Reply:
I went with `__bsCondition`